### PR TITLE
[TT-7431] Configure golangci-lint enabled/disables linters

### PR DESCRIPTION
**Pull request by titpetric**
## Description

Changes configuration of golangci-lint.

## Related Issue

https://tyktech.atlassian.net/browse/TT-7431

## Motivation and Context

Reduce verbosity of linter failures due to poorly configured linter rules.

## How This Has Been Tested

`golangci-lint run --fix ./...` - updates typoes in codebase (commit 2/2) :)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4594/merge
Commit:  
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3793639355)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4594/merge
Commit: c020b9356aa4c064c32c35327f4cb488cc4d387f 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3793671784)

**Review by jeffy-mathew — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4594/merge
Commit: d618d413c95beb4e6983452a2f4a4bb7d27b18fc 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3793786733)

File: .golangci.yml [hunk 1]
```diff
@@ -64,3 +64,40 @@ run:
   # Mainly related to generics support since go1.18.
   # Default: use Go version from the go.mod file, fallback on the env var `GOVERSION`, fallback on 1.18
   go: '1.16'
+
+linters:
+  disable:
+    - varnamelen
```
Comment (tbuchaillot) [inline .golangci.yml:70]: @titpetric can we add some comments on why we are disabling those?

Comment (titpetric) [reply .golangci.yml:70]: @tbuchaillot all in JIRA ticket + the RFC on confluence :)

**Review by tbuchaillot — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 6a1a5618822a6353f30f4aab8f98361f338f2bf4 [TT-7431] Configure golangci-lint enabled/disables linters (#4594)



## Description


Changes configuration of golangci-lint.

## Related Issue






https://tyktech.atlassian.net/browse/TT-7431

## Motivation and Context


Reduce verbosity of linter failures due to poorly configured linter
rules.

## How This Has Been Tested





`golangci-lint run --fix ./...` - updates typoes in codebase (commit
2/2) :)

## Screenshots (if appropriate)

## Types of changes



- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
functionality to change)
- [x] Refactoring or add test (improvements in base code or adds test
coverage to functionality)

## Checklist





- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning
why it's required
- [ ] I would like a code coverage CI quality gate exception and have
explained why

Co-authored-by: Tit Petric <tit@tyk.io>
Triggered by: push (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3794534830)

**Pull request merged**